src/app/ReadClient.cpp

```diff
@@ -485,6 +485,15 @@ CHIP_ERROR ReadClient::GenerateDataVersionFilterList(DataVersionFilterIBs::Build
 void ReadClient::OnActiveModeNotification()
 {
     VerifyOrDie(mpImEngine->InActiveReadClientList(this));
+
+    // Note: this API only works when issuing subscription via SendAutoResubscribeRequest, when SendAutoResubscribeRequest is
+    // called, either mEventPathParamsListSize or mAttributePathParamsListSize is not 0.
+    VerifyOrDie(mReadPrepareParams.mEventPathParamsListSize != 0 || mReadPrepareParams.mAttributePathParamsListSize != 0);
```

Comment:
Hold on a sec.  Why is this a VerifyOrDie?  Why should we _crash_ if we happen to have some random subscription that is not supposed to be auto-resubscribed?

Stopping here; please make the logic in this PR actually make sense before I spend more time looking at it.

Reply:
yes, i agree, if we have random subscription without auto, we should do nothing here.

Reply:
update with VerifyOrReturn here, thanks